[Carousel] Navigation fails on a Carousel item selection from the Content Tree

richardhand opened the pull request:
- introduces `cmp-panelcontainer-navigated` event.
- disables panel container item editable overlays that aren’t the active one.
- adds content tree listeners to navigate to and select panels on change.

<!--
Before making a PR please make sure to read our contributing guidelines
https://github.com/Adobe-Marketing-Cloud/aem-core-wcm-components/blob/master/CONTRIBUTING.md

For issue references: Add a comma-separated list of a [closing word](https://help.github.com/articles/closing-issues-via-commit-messages/)
followed by the ticket number fixed by the PR. It should be underlined in the preview if done correctly.
-->

| Q                        | A <!--(Can use an emoji 👍) -->
| ------------------------ | ---
| Fixed Issues?            | Fixes #285 <!-- remove the (`) quotes to link the issues -->
| Patch: Bug Fix?          | 👍
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      | 
| Documentation Provided   |  (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

<!-- Describe your changes below in as much detail as possible -->

jckautzmann commented on content/src/content/jcr_root/apps/core/wcm/sandbox/commons/ui/clientlibs/panelcontainer/js/PanelContainer.js:
```diff
@@ -164,6 +176,15 @@
+         * @event CQ.CoreComponents.PanelContainer#cmp-panelcontainer-navigated
+         * @param {Object} event Event object.
+         * @param {String} event.id The Panel Container ID (path)
+         * @param {Number} event.index Index of the navigated to Panel Container item
```
The description is a bit unclear. Maybe something like: 'Index of the Panel Container item that is navigated to'?

jckautzmann commented on content/src/content/jcr_root/apps/core/wcm/sandbox/commons/ui/clientlibs/panelcontainer/js/contentTree.js:
```diff
@@ -0,0 +1,141 @@
+     */
+
+    /**
+     * Sync Panel Container child {@link Granite.author.Editable} overlay disabled states on navigate,
```
this sentence is not clear to me...

richardhand commented:
@jckautzmann - addresses your feedback in https://github.com/Adobe-Marketing-Cloud/aem-core-wcm-components/pull/286/commits/f4495217fca59274397e39a4c3270fc2de131529, is it a little clearer?

richardhand commented on content/src/content/jcr_root/apps/core/wcm/sandbox/commons/ui/clientlibs/panelcontainer/js/PanelContainer.js:
```diff
@@ -20,6 +20,7 @@
     var GET_DATA_SUFFIX = ".model.json";
     var POST_SUFFIX = ".childreneditor.html";
     var MESSAGE_ID = "cmp.panelcontainer";
+    var NAVIGATE_DELAY = 200;
```
@jckautzmann, @vladbailescu - this timeout is a little brittle, it's an approximation on the time to update/navigate the UI component after messaging the content frame. Without receiving a return message I don't see any other approach. Unless you have an idea?

richardhand commented on content/src/content/jcr_root/apps/core/wcm/sandbox/commons/ui/clientlibs/panelcontainer/js/PanelContainer.js:
```diff
@@ -62,19 +63,30 @@
+                        id: data.id,
+                        index: data.index
+                    });
+                }, NAVIGATE_DELAY);
```
@vladbailescu, @jckautzmann - this delay is a little brittle, it's an approximation on the time to update/navigate the UI component after posting the message to the content frame. Without receiving a return message I don't see a better approach unless you have an idea?

jckautzmann replied:
@richardhand As you pointed, we could trigger a `navigationDone` event in the Carousel/Tabs js logic when the navigation is done and bind a handler here to listen to this event. I don't see other options.

kwin commented on content/src/content/jcr_root/apps/core/wcm/sandbox/commons/ui/clientlibs/panelcontainer/js/PanelContainer.js:
```diff
@@ -62,19 +63,30 @@
-                });
+                };
+
+                Granite.author.ContentFrame.postMessage(MESSAGE_ID, data);
```
Granite.author is given via `ns`
